src/hotspot/share/gc/g1/g1BarrierSet.inline.hpp

```diff
@@ -68,7 +68,7 @@ inline void G1BarrierSet::write_ref_field_pre(T* field) {
 }
 
 template <DecoratorSet decorators, typename T>
-inline void G1BarrierSet::write_ref_field_post(T* field, oop new_val) {
+inline void G1BarrierSet::write_ref_field_post(T* field) {
```

Comment:
It's safe because card-scanning is always prepared for "uninteresting" oops.

It's used only by Access API; its counterpart in interpreter/c1/c2 checks for null and cross-region additionally. It's not obvious whether the discrepancy is intentional or not though.

Reply:
Yeah, it looks okay.  Not a hot path I think, so the additional optimization probably isn't important.